projects/packages/forms/src/class-jetpack-forms.php

```diff
@@ -77,6 +77,16 @@ public static function is_feedback_dashboard_enabled() {
 	 * @return boolean
 	 */
 	public static function is_legacy_menu_item_retired() {
-		return apply_filters( 'jetpack_forms_retire_legacy_menu_item', false );
+
+		$default                      = false; // don't retire the legacy menu item by default
+		$largest_legacy_connection_id = 245807300; // the connection ID after which the legacy menu item is retired
+
+		$connection_id = defined( 'IS_WPCOM' ) && IS_WPCOM ? get_current_blog_id() : \Jetpack_Options::get_option( 'id' );
+
+		if ( $connection_id > $largest_legacy_connection_id ) {
+			$default = true;
+		}
+
+		return apply_filters( 'jetpack_forms_retire_legacy_menu_item', $default );
```

Comment:
@enejb -

1. For clarity, how did we determine which exactly blog id to use here (`245807300`)?

2. For determining WPCOM environment, I know there are several ways to do that, including use of the `IS_WPCOM` constant as we do here, but we also have some built in classes and methods like `( new Host() )->is_wpcom_simple()`. Does it matter what we use for that check?

Reply:
For clarity, I just created a new blog on simple sites, and it has ID of 245807828, which is about 528 (?) ids ahead of the one in this condition. I assume that's just a moving target.

Reply:
> For clarity, how did we determine which exactly blog id to use here
I did a query for the last blog_id in the site lists table. `return get_sites( ['order' => 'DESC', 'orderby' => 'id', 'number' => 4 ]);` and then padded it. We create new sites all the time so this number is always increasing.